core/lib/utils/app_logger.dart

```diff
@@ -1,70 +1,201 @@
+  }
+}
+
+void _internalLog(
+  String? message, {
+  required Level level,
+  Object? exception,
+  StackTrace? stackTrace,
+  Map<String, dynamic>? extras,
+  bool webConsoleEnabled = false,
+}) {
+  final rawMessage = _buildRawMessage(message, exception, extras, stackTrace);
+  final formattedMessage = _formatMessage(level, rawMessage);
+
+  if (webConsoleEnabled && PlatformInfo.isWeb) {
+    _printWebConsole(level, formattedMessage);
   } else {
-    switch (level) {
-      case Level.error:
-        logsStr = '[ERROR] $logsStr';
-        break;
-      default:
-        break;
-    }
+    _debugPrint(formattedMessage);
   }
+}
+
+String _buildRawMessage(
+  String? message,
+  Object? exception,
+  Map<String, dynamic>? extras,
+  StackTrace? stackTrace,
+) {
+  final parts = <String>[];
+  if (message?.isNotEmpty == true) parts.add(message!);
+  if (exception != null) parts.add('exception: $exception');
+  if (extras != null && extras.isNotEmpty) parts.add('extras: $extras');
+  if (stackTrace != null) parts.add('stackTrace: $stackTrace');
+  return parts.join(' | ');
+}
+
+
+String _formatMessage(Level level, String raw) {
+  return PlatformInfo.isWeb
+      ? _applyWebColor(level, raw)
+      : _applyMobileFormat(level, raw);
+}
+
+void _debugPrint(String formatted) {
+  if (!BuildUtils.isDebugMode) return;
+
   // ignore: avoid_print
-  print('[TwakeMail] $logsStr');
+  print('$appLogName $formatted');
 }
 
-void logError(String? value) => log(value, level: Level.error);
+void _printWebConsole(Level level, String value) {
+  switch (level) {
+    case Level.error:
+    case Level.wtf:
+      html.window.console.error('$appLogName $value');
+      break;
+    case Level.warning:
+      html.window.console.warn('$appLogName $value');
+      break;
+    case Level.info:
+      html.window.console.info('$appLogName $value');
+      break;
+    case Level.debug:
+    case Level.verbose:
+      html.window.console.debug('$appLogName $value');
+      break;
+  }
+}
 
-// Take from: https://flutter.dev/docs/testing/errors
-void initLogger(VoidCallback runApp) {
-  runZonedGuarded(() async {
-    WidgetsFlutterBinding.ensureInitialized();
-    FlutterError.onError = (details) {
-      FlutterError.dumpErrorToConsole(details);
-      logError('AppLogger::initLogger::runZonedGuarded:FlutterError.onError: ${details.stack.toString()}');
-    };
-    runApp.call();
-  }, (error, stack) {
-    logError('AppLogger::initLogger::runZonedGuarded:onError: $error | stack: $stack');
-  });
+void logError(
+  String? message, {
+  Object? exception,
+  StackTrace? stackTrace,
+  Map<String, dynamic>? extras,
+  bool webConsoleEnabled = false,
+}) {
+  _internalLog(
+    message,
+    level: Level.error,
+    exception: exception,
+    stackTrace: stackTrace,
+    extras: extras,
+    webConsoleEnabled: webConsoleEnabled,
+  );
 }
 
-class _Dispatcher extends ValueNotifier<String> {
-  _Dispatcher(String value) : super(value);
+void logWTF(
```

Comment:
```suggestion
void logCritical(
```

To align method name with the doc?